### PR TITLE
Added logging on HTTP exception

### PR DESCRIPTION
**Pull request by Nyholm**
My issue is that Im using Flysystem, which is using our adapter, which is using S3. 

Im 3 layers down in abstractions and I know S3 throws an exceptions that is caught in Flysystem. I want to log that exception somehow.

### CODE REVIEW

**Comment by Nyholm** [comment]
The first commit is more interesting to review, The other just fixed the tests and CS

File: src/Core/src/Exception/Http/HttpExceptionTrait.php [hunk 1]
```diff
@@ -68,6 +69,13 @@ public function __construct(ResponseInterface $response)
             }
         }
 
+        $logger->error($message, [
```
Comment (jderusse) [inline src/Core/src/Exception/Http/HttpExceptionTrait.php:72]: I'm not a big fan of loging in an exception's constructor :/

Comment (Nyholm) [reply src/Core/src/Exception/Http/HttpExceptionTrait.php:72]: That is fair.. But where else can I use it?

Comment (Nyholm) [reply src/Core/src/Exception/Http/HttpExceptionTrait.php]: Oh, In the Response class of course =)

**Comment by Nyholm** [comment]
PR is updated. 

Thank you @jderusse for the review. The PR is way cleaner and better now

File: src/Core/src/Response.php [hunk 2]
```diff
@@ -61,10 +62,16 @@ class Response
...
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(ResponseInterface $response, HttpClientInterface $httpClient, LoggerInterface $logger)
     {
         $this->httpResponse = $response;
         $this->httpClient = $httpClient;
+        $this->logger = $logger;
```
Comment (jderusse) [inline src/Core/src/Response.php:74]: what's about `$ths->logger = $logger ?? new NullLogger()` and remove code from Tests classes

Comment (Nyholm) [reply src/Core/src/Response.php:74]: I was considering that, but the only win is that test classes looks a bit cleaner. 

There is no point in doing that check on runtime.

**Comment by Nyholm** [comment]
Thank you for the reviews and suggestions.

**Comment by temp** [comment]
@Nyholm Is it expected to log a 404 as an error (in Response::getResolveStatus()).
I'm using async-aws with flysystem, and on each missing resource for $flysystem->has(...) an error log message is written, which results in a sentry issue here... :-D
I'd say a 404 for an existance-check is a valid answer, not an error.

**Comment by Nyholm** [comment]
Good question. I’ve seen the same thing in my logs. 

Should we log it as “info” instead?

**Comment by temp** [comment]
Yes, would be fine for me.

**Comment by Nyholm** [comment]
Do you mind submitting a PR?

**Comment by Nyholm** [comment]
Only 404 seams like the best way to move forward now.

**Pull request merged**